neurodsp/sim/params.py

```diff
@@ -0,0 +1,589 @@
+"""Simulation parameter related objects.
+
+Notes:
+- once we hit minimum python of 3.9, dictionary merging can by updated to use |
+"""
+
+from copy import deepcopy
+
+from neurodsp.sim.update import ParamIter, ParamSampler
+
+###################################################################################################
+###################################################################################################
+
+class SimParams():
+    """Object for managing simulation parameters.
+
+    Parameters
+    ----------
+    n_seconds : float
+        Simulation time, in seconds.
+    fs : float
+        Sampling rate of simulated signal, in Hz.
+
+    Attributes
+    ----------
+    base : dict
+        Dictionary of base parameters, common across all parameter definitions.
+    params : dict
+        Dictionary of created simulation parameter definitions.
+    """
+
+    def __init__(self, n_seconds=None, fs=None):
+        """Initialize SimParams object."""
+
+        self.n_seconds = n_seconds
+        self.fs = fs
+
+        self._params = {}
+
+
+    def __getitem__(self, label):
+        """Make object subscriptable, to access stored simulation parameters.
+
+        Parameters
+        ----------
+        label : str
+            Label to access simulation parameters from `params`.
+        """
+
+        return {**self.base, **self._params[label]}
```

Comment:
If user has to initialize object with n_seconds and fs, why is this return in base? Is it so the dict can be quickly unpacked into a sim func?

Reply:
Yeh,  "base" is just a shorthand to get the common parameters of n_seconds & fs (that are defined across all functions), and return with the specific parameters of some definition. Base parameters are included in order to pass this all into a sim funtion